nevergrad/optimization/base.py

```diff
@@ -736,3 +749,25 @@ def __eq__(self, other: tp.Any) -> tp.Any:
             if self._config == other._config:
                 return True
         return False
+
+
+def _constraint_solver(parameter: p.Parameter, budget: int) -> p.Parameter:
+    """Runs a suboptimization to solve the parameter constraints"""
+    parameter_without_constraint = parameter.copy()
+    parameter_without_constraint._constraint_checkers.clear()
+    opt = registry["OnePlusOne"](parameter_without_constraint, num_workers=1, budget=budget)
+    opt._constraints_manager.max_trials = 1
+    for _ in range(budget):
+        cand = opt.ask()
+        # Our objective function is minimum for the point the closest to
+        # the original candidate under the constraints.
+        penalty = cand._constraint_penalties()
```

Comment:
isnt there no more constraint? I don't understand how this could work, you probably need to get the state, set the state to a parameter with the loss, and then check for penalty

Reply:
Actually simper:
```python
penalty = sum(utils.float_penalty(func(cand.value)) for func in parameter._constraint_checkers)
```

And we can just delete `_constraint_penalties`

Reply:
Thx!